cardano-node/src/Cardano/Common/Parsers.hs

```diff
@@ -58,19 +60,80 @@ import           Cardano.Config.Types
 cliTracingParser :: Parser (Last TraceOptions)
 cliTracingParser = Last . Just <$> parseTraceOptions Opt.hidden
 
--- | The product parser for all the CLI arguments.
-nodeCliParser :: Parser NodeCLI
-nodeCliParser = do
+
+nodeProtocolModeParser  :: Parser NodeProtocolMode
+nodeProtocolModeParser = nodeRealProtocolModeParser <|> nodeMockProtocolModeParser
+
+nodeMockProtocolModeParser :: Parser NodeProtocolMode
+nodeMockProtocolModeParser = subparser
+                           (  commandGroup "Execute node with a mock protocol."
+                           <> metavar "run-mock"
+                           <> command "run-mock"
+                                (MockProtocolMode
+                                  <$> info
+                                        (nodeMockParser <**> helper)
+                                        (progDesc "Execute node with a mock protocol."))
+                           )
+nodeRealProtocolModeParser :: Parser NodeProtocolMode
+nodeRealProtocolModeParser = subparser
+                           (  commandGroup "Execute node with a real protocol."
+                           <> metavar "run"
+                           <> command "run"
+                                (RealProtocolMode
+                                  <$> info
+                                        (nodeRealParser <**> helper)
+                                        (progDesc "Execute node with a real protocol." ))
+                           )
+
+
+-- | The mock protocol parser.
+nodeMockParser :: Parser NodeMockCLI
+nodeMockParser = do
   -- Filepaths
   topFp <- parseTopologyFile
   dbFp <- parseDbPath
   genFp <- parseGenesisPath
   delCertFp <- optional parseDelegationCert
   sKeyFp <- optional parseSigningKey
-  socketFp <- parseSocketDir
+  socketFp <- parseSocketDir <|> parseSocketPath
+
+  -- NodeConfiguration filepath
+  nodeConfigFp <- parseConfigFile
 
   -- Node Address
   nAddress <- parseNodeAddress
+
+  -- TraceOptions
+  traceOptions <- cliTracingParser
+
+
+  pure $ NodeMockCLI
+           (MiscellaneousFilepaths
+              (TopologyFile topFp)
+              (DbFile dbFp)
+              (GenesisFile genFp)
+              (DelegationCertFile <$> delCertFp)
+              (SigningKeyFile <$> sKeyFp)
```

Comment:
Here's an example. I think none of the mock protocols use these 3 options.

Reply:
Removed these three from the parser.